mathesar/tests/rpc/columns/test_c_base.py

```diff
@@ -144,23 +100,24 @@ def mock_connect(_database_id, user):
         else:
             raise AssertionError('incorrect parameters passed')
 
-    def mock_column_alter(_table_oid, _column_data_list, conn):
-        if _table_oid != table_oid or _column_data_list != column_data_list:
-            raise AssertionError('incorrect parameters passed')
-        return 1
-
     monkeypatch.setattr(columns.base, 'connect', mock_connect)
-    monkeypatch.setattr(columns.base, 'alter_columns_in_table', mock_column_alter)
+    mocked_exec_msar_func.fetchone.return_value = [1]
     actual_result = columns.patch(
         column_data_list=column_data_list,
         table_oid=table_oid,
         database_id=database_id,
         request=request
     )
+    call_args = mocked_exec_msar_func.call_args_list[0][0]
+    transformed_column_data = [
+        _transform_column_alter_dict(column) for column in column_data_list
```

Comment:
Generally, I'd prefer avoiding using this function in this test. The reason is that it makes it harder to modify the underlying function (e.g., making it so that we don't even need the transform function), since it breaks this test for unrelated reasons.

Reply:
I see your point @mathemancer, but, for the case you're describing I think I would prefer if the test _did_ break, since the way the function under test used to work has changed.

If we _were_ to not use these functions in these tests however, here are some of the possible ways I see to achieve that:
1. Not checking `assert call_args[3] == json.dumps(transformed_column_data)` (Doesn't seem ideal).
2. Hardcoding the result from the function in the test (Test would still fail if the behavior of the underlying function changes).
3. Using the transformed result as an argument to API call in the test (Test fails since the underlying function expects the argument to be not transformed).

Reply:
I prefer (2).

- It clearly specifies the actual form that needs to be made to the DB layer.
- It verifies that no changes in the transformation have happened unintentionally, for example if someone were to introduce a bug in the `_transform_column_alter_dict` function itself.
- It detaches the test from the internal implementation of the function being tested as much as possible.

I agree the test should break if we align the API `column_data_list` with the form of the SQL layer `column_data_list`, but we _don't_ want it to break because (for example) the function being used in the test is now missing, which it would in that case with this test as currently written. We want it to break because we expect the input and output to be different, but they're now the same. I.e., we want the test assertion to say "Hey! this JSON isn't what I expected", not "Hey! There's a private function missing!".

Hopefully that makes sense.